x/programs/rust/wasmlanche_sdk/src/types.rs

```diff
@@ -35,45 +32,68 @@ impl IntoIterator for Address {
     }
 }
 
-/// A struct representing a fixed length of 32 bytes.
-/// This can be used for passing strings to the host. It caps the string at 32 bytes,
-#[derive(Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Debug)]
-pub struct Bytes32([u8; Self::LEN]);
-impl Bytes32 {
-    pub const LEN: usize = 32;
-    #[must_use]
-    pub fn new(bytes: [u8; Self::LEN]) -> Self {
-        Self(bytes)
-    }
-    #[must_use]
-    pub fn as_bytes(&self) -> &[u8] {
-        &self.0
+/// A struct representing variable length bytes.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub struct Bytes<const N: usize>([u8; N]);
+
+impl<const N: usize> Serialize for Bytes<N> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        // Serialize the array as a sequence of bytes
+        serializer.serialize_bytes(&self.0)
     }
 }
 
-/// Implement the Display trait for Bytes32 so that we can print it.
-/// Enables `to_string()` on Bytes32.
-impl std::fmt::Display for Bytes32 {
-    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        // Find the first null byte and only print up to that point.
-        let null_pos = self.0.iter().position(|&b| b == b'\0').unwrap_or(Self::LEN);
-        String::from_utf8_lossy(&self.0[..null_pos]).fmt(f)
+impl<'de, const N: usize> Deserialize<'de> for Bytes<N> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct BytesVisitor<const N: usize>;
+
+        impl<'de, const N: usize> Visitor<'de> for BytesVisitor<N> {
+            type Value = Bytes<N>;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str(&format!("{N} bytes"))
+            }
+
+            fn visit_seq<A>(self, mut seq: A) -> Result<Bytes<N>, A::Error>
+            where
+                A: SeqAccess<'de>,
+            {
+                let mut bytes = [0u8; N];
+                for (i, byte) in bytes.iter_mut().enumerate() {
+                    *byte = seq.next_element::<u8>()?.ok_or_else(|| {
+                        de::Error::invalid_length(i, &format!("{N} bytes").as_str())
+                    })?;
+                }
+                Ok(Bytes(bytes))
+            }
+        }
+
+        deserializer.deserialize_bytes(BytesVisitor::<N>)
     }
 }
 
-impl From<String> for Bytes32 {
-    fn from(value: String) -> Self {
-        let mut bytes: [u8; Self::LEN] = [0; Self::LEN];
-        bytes[..value.len()].copy_from_slice(value.as_bytes());
+impl<const N: usize> Bytes<N> {
```

Comment:
@samliok now that bytes are generic I think we can plugin the code you are working on?